manifests/rule/remote.pp

```diff
@@ -303,12 +305,48 @@
           $_stream_driver_permitted_peers = "*.${facts['domain']}"
           notify { "TLS StreamDriverPermittedPeers ${name}":
             message  => ("rsyslog::rule::remote ${_notify_msg}"),
-            loglevel => 'warning'
+            loglevel => 'warning',
           }
         }
       }
     }
 
+    # Basic validation for the action queue parameters
+    if $queue_size {
+      # First check the queue size, which should not be less than 100 per the docs
+      if $queue_size < 100 {
+        # Warn the user about an errant configuration, but don't fail as RSyslog will still run when setup this way
+        notify { "Invalid queue_size specified for ${name}":
+          message  => "Action queue size for ${name}: ${queue_size} is less than 100 and can have adverse effects on RSyslog",
```

Comment:
```suggestion
          message  => "Action queue size for ${name}: ${queue_size} is less than 100 and can have adverse effects on RSyslog. To disable this message set rsyslog::rule::remote::queue_validation_log_level to 'debug' in Hiera or your ENC",
```

Reply:
Interesting solution; I didn't know know you could do that with `notify`.